 config.gcc |    5 +++++
 1 files changed, 5 insertions(+)

--- gcc/config.gcc
+++ gcc/config.gcc
@@ -902,6 +902,11 @@
 	tmake_file="i386/t-i386elf t-svr4"
 	use_fixproto=yes
 	;;
+x86_64-*-elf*)
+	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h i386/i386elf.h i386/x86-64.h"
+	tmake_file="i386/t-i386elf t-svr4"
+	use_fixproto=yes
+	;;
 i[34567]86-sequent-ptx4* | i[34567]86-sequent-sysv4*)
 	if test x$gas = xyes
 	then
