diff -ru gcc-4.2.2/gcc/dwarf2out.c gcc-4.2.2.final/gcc/dwarf2out.c
--- gcc-4.2.2/gcc/dwarf2out.c	2007-09-24 10:17:10.000000000 +0100
+++ gcc-4.2.2.final/gcc/dwarf2out.c	2008-04-22 16:12:21.000000000 +0100
@@ -281,9 +281,9 @@
 /* Offsets recorded in opcodes are a multiple of this alignment factor.  */
 #ifndef DWARF_CIE_DATA_ALIGNMENT
 #ifdef STACK_GROWS_DOWNWARD
-#define DWARF_CIE_DATA_ALIGNMENT (-((int) UNITS_PER_WORD))
+#define DWARF_CIE_DATA_ALIGNMENT (-((int) 4/*UNITS_PER_WORD*/))
 #else
-#define DWARF_CIE_DATA_ALIGNMENT ((int) UNITS_PER_WORD)
+#define DWARF_CIE_DATA_ALIGNMENT ((int) 4/*UNITS_PER_WORD*/)
 #endif
 #endif
 
diff -ru gcc-4.2.2/gcc/function.c gcc-4.2.2.final/gcc/function.c
--- gcc-4.2.2/gcc/function.c	2007-09-01 16:28:30.000000000 +0100
+++ gcc-4.2.2.final/gcc/function.c	2008-04-22 16:12:21.000000000 +0100
@@ -2887,7 +2887,11 @@
 			   BLOCK_OP_NORMAL);
 	}
       else
-	emit_move_insn (dest, src);
+      {
+	rtx insn = emit_move_insn (dest, src);
+        RTX_FRAME_RELATED_P (insn) = 1;
+      }
+      
     }
 
   if (to_conversion)
