--- /home/james/pedigree-new/src/user/applications/bash/bash-3.2/shell.c	2006-05-17 13:46:54.000000000 +0100
+++ ../shell.c	2007-09-16 13:50:01.000000000 +0100
@@ -334,9 +334,9 @@
   int saverst;
 #endif
   volatile int locally_skip_execution;
   volatile int arg_index, top_level_arg_index;
-#ifdef __OPENNT
+#if defined(__OPENNT) || defined(NO_MAIN_ENV_ARG)
   char **env;
 
   env = environ;
 #endif /* __OPENNT */
